### DISCUSSION TITLE
Use ses without unsafe-eval in CSP

### DISCUSSION THREAD

**Discussion by imolorhe** [endojs/endo#1932] [Q&A]
Is it possible to use `ses` without adding `unsafe-eval` to a site's CSP?

**Comment by Jack-Works** [comment]
for lockdown, the answer is yes, but no (see #1281).

for compartment, the answer is no. if you need a no-eval version of the compartment, you need a compiler (https://github.com/DimensionDev/aot-secure-ecmascript/tree/main/packages/static-module-record-swc/src) and another virtualized module runtime (https://github.com/DimensionDev/aot-secure-ecmascript/tree/main/packages/compartment).
